micrometer-core/src/test/java/io/micrometer/core/instrument/logging/LoggingMeterRegistryTest.java

```diff
@@ -43,6 +46,24 @@ void defaultMeterIdPrinter() {
         assertThat(printer.id()).isEqualTo("my.gauage{tag-1=tag-2}");
     }
 
+    @Test
+    void publishUsesProvidedSinkFromConstructors() {
+        String expectedString = "my.gauage{tag-1=tag-2} value=1";
+        // simulating a mock
+        Consumer<String> sink = s -> assertThat(s).isEqualTo(expectedString);
```

Comment:
I believe from looking at the implementation that this assertion will be run, but even if it weren't run, this test would still pass. It would be best to write the test in a way that it fails if the consumer is NOT run.

Reply:
micrometer-core has mockito in test scope: https://github.com/micrometer-metrics/micrometer/blob/9fe11a524a7026b175093cb02227561ec575c268/micrometer-core/build.gradle#L148